pkg/s3select/csv/record.go

```diff
@@ -61,30 +61,28 @@ func (r *Record) Set(name string, value *sql.Value) error {
 	return nil
 }
 
-// MarshalCSV - encodes to CSV data.
-func (r *Record) MarshalCSV(fieldDelimiter rune) ([]byte, error) {
-	buf := new(bytes.Buffer)
-	w := csv.NewWriter(buf)
-	w.Comma = fieldDelimiter
-	if err := w.Write(r.csvRecord); err != nil {
-		return nil, err
+// WriteCSV - encodes to CSV data.
+func (r *Record) WriteCSV(w io.Writer, fieldDelimiter rune) error {
+	csvW := csv.NewWriter(w)
+	csvW.Comma = fieldDelimiter
+	if err := csvW.Write(r.csvRecord); err != nil {
+		return err
 	}
-	w.Flush()
-	if err := w.Error(); err != nil {
-		return nil, err
+	csvW.Flush()
+	if err := csvW.Error(); err != nil {
+		return err
 	}
 
-	data := buf.Bytes()
-	return data[:len(data)-1], nil
+	return nil
```

Comment:
Current code returns `[]byte` contains no trailing newline whereas this modification writes CSV record with newline.  Have you checked `OutputSerialization.CSV.RecordDelimiter` with other than `\n`?

Reply:
yes, this case has been processed at https://github.com/yaozongyou/minio/blob/speed-up-select-on-csv/pkg/s3select/select.go#L329
the newline is truncated and RecordDelimiter is appended.